odoo/orm/fields.py

```diff
@@ -1558,18 +1557,36 @@ def write(self, records: BaseModel, value: typing.Any) -> None:
     # Cache management methods
     #
 
-    def _get_cache(self, env: Environment) -> MutableMapping[IdType, typing.Any]:
+    def _get_cache(self, env: Environment, *, to_write: bool = False) -> MutableMapping[IdType, typing.Any]:
         """ Return the field's cache, i.e., a mutable mapping from record id to
         a cache value.  The cache may be environment-specific.  This mapping is
         the way to retrieve a field's value for a given record.
 
         Calling this function multiple times, always returns the same mapping
         instance for a given environment, unless the transaction was entirely
         invalidated.
+
+        When calling the first time, the field access is checked for the model.
+        Sometimes, you may want to retrieve a cache even if you don't have
+        access, for example when updating values. In that case, set `to_write`
+        flag: in case you don't have access, it will return the cache of the
+        sudoed environment instead.
         """
         try:
             return env._field_cache_memo[self]
         except KeyError:
+            if not (env.su or env[self.model_name]._has_field_access(self, 'read')):
+                # optimization: we called _has_field_access() to avoid an extra
+                # function call in _check_field_access()
+                if to_write:
+                    # updates check already for field access, giving them the
+                    # cache in sudo instead to put values in them
+                    return self._get_cache(env(su=True))
```

Comment:
I find this statement strongly misleading.  In the case of a field that "context-depends" on `uid`, the cache in `env(su=True)` is not the same cache as the one in `env`!